src/frontend/components/TimedTextCreationForm/TimedTextCreationForm.tsx

```diff
@@ -0,0 +1,163 @@
+    description:
+      'Generic error message when we fail to create a subtitle/transcript/caption track.',
+    id: 'components.TimedTextCreationForm.error',
+  },
+});
+
+const TimedTextCreationFormStyled = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  margin-top: 1rem;
+`;
+
+const ErrorMessage = styled.div`
+  color: ${colors.danger.main};
+`;
+
+interface SelectOption {
+  label: string;
+  value: string;
+}
+
+/** Props shape for the TimedTextCreationForm component. */
+export interface TimedTextCreationFormProps {
+  createTimedTextTrack: (timedtexttrack: TimedText) => void;
+  jwt: Nullable<string>;
+  mode: timedTextMode;
+}
+
+/** State shape for the TimedTextCreationForm component. */
+interface TimedTextCreationFormState {
+  error?: 'creation' | 'schema';
+  languageChoices: SelectOption[];
+  newTTLanguage?: string;
+  newTTUploadId?: TimedText['id'];
+}
+
+/**
+ * Component. Displays a form that allows the user to create a new timedtexttrack.
+ * @param createTimedTextTrack Action creator that takes a timedtexttrack to insert into the store.
+ * @param jwt The token that will be used to interact with the API.
+ * @param mode The mode of the timedtexttracks we're creating.
+ */
+export class TimedTextCreationForm extends React.Component<
+  TimedTextCreationFormProps,
+  TimedTextCreationFormState
+> {
+  constructor(props: TimedTextCreationFormProps) {
+    super(props);
+    this.state = { languageChoices: [] };
+  }
+
+  async componentDidMount() {
+    try {
+      const languageChoices = await getTimedTextTrackLanguageChoices(
+        this.props.jwt,
+      );
+      this.setState({
+        languageChoices,
+      });
+    } catch (error) {
+      this.setState({ error: 'schema' });
+    }
+  }
+
+  async createAndGoToUpload() {
+    try {
+      const newTTT = await createTimedTextTrack(
+        this.props.jwt,
+        this.state.newTTLanguage!,
+        this.props.mode,
+      );
+
+      this.props.createTimedTextTrack(newTTT);
+      this.setState({ newTTUploadId: newTTT.id });
+    } catch (error) {
+      this.setState({ error: 'creation' });
+    }
+  }
+
+  onSelectChange(
+    option: Maybe<Nullable<ValueType<SelectOption>>>,
+    { action }: ActionMeta,
+  ) {
+    if (action === 'select-option' && option && !(option instanceof Array)) {
+      this.setState({ newTTLanguage: option.value });
+    }
+  }
+
+  render() {
+    const { error, languageChoices, newTTLanguage, newTTUploadId } = this.state;
+
+    if (error === 'schema') {
+      return <Redirect push to={ERROR_COMPONENT_ROUTE('notFound')} />;
+    }
+
+    if (newTTUploadId) {
+      return (
+        <Redirect
+          push
+          to={UPLOAD_FORM_ROUTE(modelName.TIMEDTEXTTRACKS, newTTUploadId)}
+        />
+      );
+    }
+
+    return (
+      <TimedTextCreationFormStyled>
+        <FormattedMessage {...messages.addTrackLabel} />
+        <Select
+          onChange={this.onSelectChange.bind(this)}
```

Comment:
IMO binding in the constructor is more confusing and having to bind methods is one of the "bad parts" of react. I'd rather not do this until we have an actual problem with the render binding.

Hopefully Hooks are released soon and we can get rid of this problem altogether 😊

Reply:
This is a stage 3 proposal. At this point I think it will probably be implemented but I think it's a good policy to refrain from using stage 3 and below as much as possible.